articles/changes.md

```diff
@@ -0,0 +1,241 @@
+
+
+## Messages, Services
+
+### Separated namespaces
+
+In ROS 1 .msg and .srv files can have the same name but the generated code collides.
+The same is the case for the request and response parts of services.
+
+In ROS 2 the generated code uses separate namespaces to guarantee it is collision-free.
+
+
+### Optional default values in message definitions
+
+### Optional upper bounds for arrays and strings
+
+This is necessary to allow preallocation of messages with a dynamic sizes in use cases like real-time.
+
+
+### Unify duration and time types
+
+In ROS 1 the duration and time types are defined in the client libraries.
+The member names of the data structures are different in C++ (sec, nsec) and Python (secs, nsecs).
+
+In ROS 2 these types will be defines as messages and therefore be consistent across languages.
+
+
+### Remove sequence field from Header message
+
+The field has been deprecated for a long time and was not set consistently in ROS 1.
+
+
+## Client libraries
+
+### Across languages
+
+#### Notifications
+
+In ROS 1 all information about the ROS graph must be polled from the master.
+In ROS 2 changes will be published instead.
+
+
+#### Components with life cycle
+
+In ROS 1 every node usually has its own main function.
+In ROS 2 it will be recommended to subclass from a component which has a life cycle.
+
+The life cycle can be used by tools like roslaunch to start a system composed of many components in a deterministic way.
+
+
+#### Parameters, Dynamic Reconfigure
+
+In ROS 1 global parameters and node-specific dynamic reconfigure parameters are two separate concepts.
+In ROS 2 a unified approach is being used.
+It will be similar to dynamic reconfigure and a node named "global parameter server" will accept requests to set value unconditionally.
+Again while in ROS 1 all these information need to be polled for changes in ROS 2 changes will be published to notify other entities.
+
+
+#### Threading model
+
+In ROS 1 the developer can only choose between single-threaded execution or multi-threaded execution.
+In ROS 2 more granular execution models are available.
+
+
+#### Graph API
+
+In ROS 1 nodes and topics can be remapped at startup time only.
+In ROS 2 it will be possible to remap those at runtime as well as create aliases.
+
+
+### C and C++
+
+#### Support for real-time
+
+ROS 1 does not support writing real-time code but relies on external frameworks like Orocos.
+In ROS 2 it will be possible to write real-time nodes when using a proper RTOS and with carefully written user code.
+
+
+### C++
+
+#### Node vs. Nodelet
+
+In ROS 1 the API for nodes and nodelets is different and requires a decision at programming time.
+In ROS 2 it will be possible to made this a deploy-time decision.
+It will be recommended to compile each component into a shared library which can then be loaded in a separate process or shared the same process with other components.
```

Comment:
I have updated the sentence. Please check if it is clear enough now or needs further information.